model/schema/geneInteraction.yaml

```diff
@@ -50,33 +50,30 @@ classes:
   ## ASSOCIATIONS
   ## ------------
 
-  GeneToGeneAssociation:
+  GeneGeneAssociation:
     description: >-
       abstract parent class for different kinds of gene-gene or gene product
       to gene product relationships. Includes homology and interaction.
     abstract: true
     is_a: EvidenceAssociation
-    defining_slots:
-      - subject
-      - object
-    slot_usage:
-      subject:
+    attributes:
+      gene_gene_association_object:
         range: Gene
+        required: true
         description: >-
-          the subject gene in the association. If the relation is symmetric,
+          the object gene in the association. If the relation is symmetric,
           subject vs object is arbitrary. We allow a gene product to stand
           as a proxy for the gene or vice versa.
-      object:
-        range: Gene
+    slot_usage:
+      gene_association_subject:
```

Comment:
I see that "gene_association_subject" is declared in GeneGeneAssociation slot_usage and as a defining slot in GeneInteraction, but the slot itself is never actually instantiated in any class slot list. I suppose it would have gone in a more generic GeneAssociation class above GeneGeneAssociation, but that class does not exist. Do we need a generic GeneAssociation class? In any case, the slot "gene_association_subject" needs to be declared as a slot in some base class. @sierra-moxon Do we have any LinkML pipeline checks to make sure that if a slot is considered defining for a class, then the slot must have been instantiated as a slot for that class or some superclass?

Reply:
Good spot, slot definition now added.  A GeneAssociation class with the gene_association_subject slot as an attribute isn't really an option as we have different gene associations that inherit from LocationAssociation, EvidenceAssociation, and SingleReferenceAssociation.